### ISSUE TITLE
Consider switching from JSHint to ESLint

### ISSUE THREAD

**Issue by adroitwhiz** [paperjs/paper.js#1827]
Currently, JSHint is used to enforce code correctness. However, it does not enforce code style (e.g. maximum line length).

ESLint is intended to enforce both correctness and style, and so could be used as a replacement for JSHint that could do things like automatically ensure e.g. lines are <= 80 characters long so you don't have to in PR reviews.

In addition, ESLint is much better maintained and more customizable.

**Comment by lehni** [comment]
Yeah there are a lot of things we should change in regards to that. The code base was started 9 years ago, and many things haven't been modernised since then.
